app/data/generators/personal-details.js

```diff
@@ -30,12 +30,68 @@ module.exports = faker => {
     details: (rightToWorkStudy === 'Yes') ? 'I have EU settled status' : false
   } : false
 
+  // Equality and diversity
+  const diversityQuestionnaireAnswered=faker.helpers.randomize([true, true, false])
+
+  let diversityQuestionnaire
+
+  const sexInteger = faker.helpers.randomize([0, 1])
+
+  let sex
+  let ethnicGroup
+  let disabledAnswer
+  let disabilities
+
+  if (diversityQuestionnaireAnswered){
+
+    sex = { 0: "Male", 1: "Female" }[sexInteger]
+
+    ethnicGroup = faker.helpers.randomize([
+      "Asian or Asian British",
+      "Black, African, Black British or Caribbean",
+      "Mixed or multiple ethnic groups",
+      "White",
+      "Another ethnic group",
+      "Prefer not to say"
+    ])
+
+    disabledAnswer = faker.helpers.randomize(["Yes", "No", "Prefer not to say"])
+
+    disabilityCount = faker.random.number(3); // up to 3 disabilities
+
+    let disabilityChoices = [
+      "Blind",
+      "Deaf",
+      "Learning difficulty",
+      "Long-standing illness",
+      "Mental health condition",
+      "Physical disability or mobility issue",
+      "Social or communication impairment",
+      "Other"
+    ]
+    let shuffledDisabilities = disabilityChoices.sort(() => 0.5 - Math.random());
+
+    if ((disabledAnswer=="Yes") && disabilityCount){
+      disabilities = shuffledDisabilities.slice(0, disabilityCount).sort();
+    }
+
+  }
+
+
+
+
   return {
-    givenName: faker.name.firstName(),
-    familyName: faker.name.lastName(),
+    givenName: faker.name.firstName(sexInteger),
+    familyName: faker.name.lastName(sexInteger),
     dateOfBirth: faker.date.between('1958-01-01', '1998-01-01'),
     nationality,
     residency,
-    isInternationalCandidate
+    isInternationalCandidate,
+    diversityQuestionnaireAnswered,
```

Comment:
So I would normally derive this in the route/view rather than bloating the data. But as it's a prototype it's less important.

You can see PL did it with the isInternationalCandidate. Where as I would have derived it in the view by saying `nationality === 'United Kingdom'`